browser/help/faq/mitochondrial-dna/how-many-samples-are-in-each-mtdna-haplogroup-for-each-nuclear-ancestry-population.js

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import { BaseTable, Select } from '@gnomad/ui'
+
+import ancestryHaploData from './how-many-samples-are-in-each-mtdna-haplogroup-for-each-nuclear-ancestry-population.json'
+
+const codeToWord = {
+  afr: 'African/African American',
+  ami: 'Amish',
+  amr: 'Latino/Admixed American',
+  asj: 'Ashkenazi Jewish',
+  eas: 'East Asian',
+  fin: 'European (Finnish)',
+  mid: 'Middle Eastern',
+  nfe: 'European (non-Finnish)',
+  oth: 'Other',
+  sas: 'South Asian',
+}
+
+const HaplogroupAncestryTable = styled(BaseTable)`
+  margin-top: 1em;
+  margin-bottom: 1em;
+
+  td {
+    text-align: right;
+  }
+`
+
+const HaploAncestrySelect = styled(Select)`
+  margin: 1em 1em 0 0;
+`
+
+const AncestryAndHaplogroupTable = () => {
+  const [haplogSelected, setHaplogSelected] = useState('All')
+  const [ancestSelected, setAncestSelected] = useState('All')
+  const [filteredData, setFilteredData] = useState(ancestryHaploData.data)
+  const haplogroupOptions = [
+    'A',
+    'B',
+    'C',
+    'D',
+    'E',
+    'F',
+    'G',
+    'H',
+    'HV',
+    'I',
+    'J',
+    'K',
+    'L0',
+    'L1',
+    'L2',
+    'L3',
+    'L4',
+    'L5',
+    'M',
+    'N',
+    'P',
+    'R',
+    'T',
+    'U',
+    'V',
+    'W',
+    'X',
+    'Y',
+    'Z',
+  ]
+  const ancestryOptions = [
+    'African/African American',
+    'Amish',
+    'Latino/Admixed American',
+    'Ashkenazi Jewish',
+    'East Asian',
+    'European (Finnish)',
+    'Middle Eastern',
+    'European (non-Finnish)',
+    'South Asian',
+    'Other',
+  ]
+
+  useEffect(() => {
+    const newFilteredData = ancestryHaploData.data.filter(element => {
+      const splitElement = element.split(' ')
+      return (
+        (splitElement[0] === haplogSelected || haplogSelected === 'All') &&
+        (codeToWord[splitElement[1]] === ancestSelected || ancestSelected === 'All')
+      )
+    })
+    setFilteredData(newFilteredData)
+  }, [haplogSelected, ancestSelected])
+
+  return (
+    <>
+      <HaploAncestrySelect
+        id="haplogroup-selected"
+        value={haplogSelected}
+        onChange={e => {
```

Comment:
The snapshot test is a good start here, but we also have new interactions on this page that should be covered as well, i.e. the data updating when you change your selection in these `select`s. However, that would require some more setup and possibly some choices between different tools, which would expand the scope of this PR a bit more than we might want, while conversely it'd fold naturally into the work on test infrastructure I've been doing lately.

So what I think is, let's leave test that out of this PR, but after it's merged I will come back and use this interaction as a testbed for this other style of test. Then the test for this interaction can serve as an example for similar tests going forward.

Reply:
This sounds good to me.